packages/iov-core/src/signingservice.spec.ts

```diff
@@ -73,13 +73,13 @@ function makeSimpleMessagingConnection(
     start: listener => {
       // tslint:disable-next-line:no-object-mutation
       worker.onmessage = event => {
-        const responseError = parseJsonRpcError(event.data);
-        if (responseError) {
-          listener.next(responseError);
-        } else {
-          const response = parseJsonRpcResponse(event.data);
-          listener.next(response);
+        let response: JsonRpcResponse;
+        try {
+          response = parseJsonRpcErrorResponse(event.data);
+        } catch (_) {
+          response = parseJsonRpcSuccessResponse(event.data);
```

Comment:
What if this throws?

Reply:
And unhandled exception is thrown and execution stops (as before)